### PR TITLE
Fix the single_thread_pointer 

### PR DESCRIPTION
**Pull request by azimafroozeh**

### CODE REVIEW

File: src/include/duckdb/common/single_thread_ptr.hpp [hunk 1]
```diff
@@ -123,10 +125,11 @@ class single_thread_ptr {
 template <class T>
 struct _object_and_block {
 	T object;
-	uint32_t pn = 1;
+	uint32_t *pn;
```
Comment (Mytherin) [inline src/include/duckdb/common/single_thread_ptr.hpp:128]: The point of the object_and_block is that the object and the reference counter are both in the same memory region, no? Turning this into a pointer means that this is no longer the case, which adds an additional layer of indirection for accessing the reference count. This seems to defeat the point of the _object_and_block class.

Could you have a look at fixing this without changing this into a pointer?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR. I haven't tried it, but I believe this would indeed fix the problem. I do have one comment though:

Also, please use more descriptive pull request names in the future.

File: src/include/duckdb/common/single_thread_ptr.hpp (hunk 1 above)
Comment (azimafroozeh) [reply src/include/duckdb/common/single_thread_ptr.hpp]: Sure.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**